### PR TITLE
MNT: re-organize galleries under one subdir

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

This reorganizes all the galleries (examples, tutorials, plot_types) under one subdirectory.  Also renames examples->gallery so that it is the same name as the gallery it makes.  

So now sphinx-gallery looks in `/galleries/gallery`, `/galleries/tutorials/` and `/galleries/plot_types`.  

The goal here is that now that https://github.com/sphinx-gallery/sphinx-gallery/pull/1071 is in, we may have more subdirectories run through sphinx gallery (perhaps many more), and it pollutes the top-level file hierarchy to have a bunch of these.   

Note that no examples were changed by this, and there should be no difference on the doc page (except in the "writing Documentation" page where the file names have changed).  

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: doc/devel/documenting_mpl.rst [hunk 1]
```diff
@@ -21,10 +21,11 @@ the docstrings of the classes in the Matplotlib library.  Except for
 when the documentation is built.  See :ref:`writing-docstrings` below.
 
 Second, the contents of :file:`doc/plot_types`, :file:`doc/gallery` and
-:file:`doc/tutorials` are generated by the `Sphinx Gallery`_ from python
-files in :file:`plot_types/`, :file:`examples/` and :file:`tutorials/`.
-These sources consist of python scripts that have ReST_ documentation built
-into their comments.  See :ref:`writing-examples-and-tutorials` below.
+:file:`doc/tutorials` are generated by the `Sphinx Gallery`_ from python files
```
Comment (jklymak) [inline doc/devel/documenting_mpl.rst:24]: This file is the only user-facing change.

**Comment by tacaswell** [comment]
👍🏻 in principle.

**Comment by timhoffm** [comment]
"examples" is a better name than "gallery", because it's more semantic (also `galleries/Gallery` is awkward). Can we go the other way and add a http redirect from gallery to example to preserve existing links working, so that we can use "example" consistently.

**Comment by jklymak** [comment]
The change here just makes the private name the same as the public.  I'm not sure if there is a mechanism to change the public name.  If there is, changing the private is just a one-line "git mv" and a few edit in the dev docs and the conf.py. 

We could also keep the private and public names different (that was not really the point here) but I think it is needlessly confusing.

**Comment by timhoffm** [comment]
> We could also keep the private and public names different

Slight preference for doing this if you want to move forward without going into "changing the public name" right now. Having "gallery" as the internal name and url, but resulting in a page titled "Examples" which is under the menu entry "Examples" is still confusing. The only consistent naming option is all "examples".

**Comment by jklymak** [comment]
The structure is now `galleries/examples` -> `doc/gallery`.    Redirecting `doc/gallery` -> `doc/examples` via http redirect requires getting into the setup mechanisms for our web host

**Comment by ksunden** [comment]
Unfortunately #25270 conflicted the animation examples (and a couple README's)... should be relatively trivial (probably for all of the animation ones at least it is the grammatical change from `generate` to `generated` on the note about the html viewer that was made on the backport.)

Also, am I misremembering the call? I thought the conclusion of that conversation was that we preferred the term "examples" over "gallery" for the repo paths (and desired doing redirects to make the url match as "examples", rather than its current "gallery", though I can acknowledge that much may be out of scope for now).

The extra level of heirarchy is fine, and I don't mind that being called "galleries", but for those called "examples" now, keeping it so was my impression of the consensus.

**Comment by jklymak** [comment]
Sorry, I somehow messed up the last push.

**Comment by QuLogic** [comment]
It looks like `.github/workflows/tests.yml`, `.gitignore`, `.pre-commit-config.yaml` and `pyproject.toml` need updating.

**Comment by jklymak** [comment]
> It looks like `.github/workflows/tests.yml`, `.gitignore`, `.pre-commit-config.yaml` and `pyproject.toml` need updating.

Phew, we sure repeat the same exceptions in a lot of places....

**Comment by jklymak** [comment]
The pre-commit isort block doesn't hit `galleries/plot_types`.  I didn't fix that because that directory fails isort, and didn't want to sidetrack this...

File: .gitignore [hunk 2]
```diff
@@ -56,8 +56,8 @@ Thumbs.db
 
 # Things specific to this project #
 ###################################
-tutorials/intermediate/CL01.png
-tutorials/intermediate/CL02.png
+galleries/tutorials/intermediate/CL01.png
+galleries/intermediate/CL02.png
```
Comment (QuLogic) [inline .gitignore:60]:
```suggestion
galleries/tutorials/intermediate/CL02.png
```

File: .gitignore [hunk 3]
```diff
@@ -72,14 +72,14 @@ doc/plot_types
...
-examples/tests/*
-!examples/tests/backend_driver_sgskip.py
+galleries/examples/*/*.bmp
+galleries/examples/*/*.eps
+galleries/examples/*/*.pdf
+galleries/examples/*/*.png
+galleries/examples/*/*.svg
+galleries/examples/*/*.svgz
+galleries/examples/tests/*
+!galleries/examples/tests/backend_driver_sgskip.py
```
Comment (QuLogic) [inline .gitignore:82]: We can delete these two; there are no tests in `examples` any more.

**Review by QuLogic — APPROVED** [review]

**Comment by jklymak** [comment]
Thanks for everyone's patient reviews...

**Pull request merged**